src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs

```diff
@@ -0,0 +1,324 @@
+                                                                         s_localizableTitleCA2017,
+                                                                         s_localizableMessageCA2017,
+                                                                         DiagnosticCategory.Reliability,
+                                                                         RuleLevel.BuildWarning,
+                                                                         description: s_localizableDescriptionCA2017,
+                                                                         isPortedFxCopRule: false,
+                                                                         isDataflowRule: false,
+                                                                         isReportedAtCompilationEnd: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(CA1727Rule, CA1848Rule, CA2253Rule, CA2254Rule, CA2017Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.EnableConcurrentExecution();
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
+
+                if (!wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingLoggerExtensions, out var loggerExtensionsType) ||
+                    !wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingILogger, out var loggerType) ||
+                    !wellKnownTypeProvider.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftExtensionsLoggingLoggerMessage, out var loggerMessageType))
+                {
+                    return;
+                }
+
+                context.RegisterOperationAction(context => AnalyzeInvocation(context, loggerType, loggerExtensionsType, loggerMessageType), OperationKind.Invocation);
+            });
+        }
+
+        private void AnalyzeInvocation(OperationAnalysisContext context, INamedTypeSymbol loggerType, INamedTypeSymbol loggerExtensionsType, INamedTypeSymbol loggerMessageType)
+        {
+            var invocation = (IInvocationOperation)context.Operation;
+
+            var methodSymbol = invocation.TargetMethod;
+            var containingType = methodSymbol.ContainingType;
+            bool usingLoggerExtensionsTypes = false;
+
+            if (containingType.Equals(loggerExtensionsType, SymbolEqualityComparer.Default))
+            {
+                usingLoggerExtensionsTypes = true;
+                context.ReportDiagnostic(invocation.CreateDiagnostic(CA1848Rule, methodSymbol.ToDisplayString(GetLanguageSpecificFormat(invocation)), methodSymbol.Name));
```

Comment:
This is passing two arguments to `CreateDiagnostic`, while the message of this rule has only one placeholder.